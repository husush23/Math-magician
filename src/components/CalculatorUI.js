import { useState } from 'react';
import calculate from '../logic/calculate';

function CalculatorUI() {
  const [values, setValues] = useState({
    total: null,
    next: null,
    operation: null,
  });
  function handleClick(e) {
    setValues(calculate(values, e.target.innerText));
  }

  return (
    <div className="calculator">
      <div className="output" title="Output">
        <span>{values.total && values.total}</span>
        <span>{values.operation && values.operation}</span>
        <span>{values.next && values.next}</span>
      </div>
      <div className="keyboard">
        <button
          type="submit"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          AC
        </button>
        <button
          type="submit"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          +/-
        </button>
        <button
          type="submit"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          %
        </button>
        <button
          type="submit"
          className="side"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          รท
        </button>
        <button
          type="submit"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          7
        </button>
        <button
          type="submit"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          8
        </button>
        <button
          type="submit"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          9
        </button>
        <button
          type="submit"
          className="side"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          x
        </button>
        <button
          type="submit"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          4
        </button>
        <button
          type="submit"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          5
        </button>
        <button
          type="submit"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          6
        </button>
        <button
          type="submit"
          className="side"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          -
        </button>
        <button
          type="submit"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          1
        </button>
        <button
          type="submit"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          2
        </button>
        <button
          type="submit"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          3
        </button>
        <button
          type="submit"
          className="side"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          +
        </button>
        <button
          type="submit"
          className="span"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          0
        </button>
        <button
          type="submit"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          .
        </button>
        <button
          type="submit"
          className="side"
          onClick={(e) => {
            handleClick(e);
          }}
        >
          =
        </button>
      </div>
    </div>
  );
}
export default CalculatorUI;
