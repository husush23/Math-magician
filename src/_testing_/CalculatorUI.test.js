import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CalculatorUI from '../components/CalculatorUI';

describe('Calculator', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<CalculatorUI />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Adds correctly based on input', () => {
    render(<CalculatorUI />);
    const output = screen.getByTestId('calculator-output');
    expect(output.textContent).toBe('');

    const addButton = screen.getByText('+');
    const equalsButton = screen.getByText('=');

    fireEvent.click(screen.getByText('9'));
    fireEvent.click(addButton);
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(equalsButton);
    expect(output.textContent).toBe('12');
  });

  test('Subtracts correctly based on input', () => {
    render(<CalculatorUI />);
    const output = screen.getByTestId('calculator-output');
    expect(output.textContent).toBe('');

    const minusButton = screen.getByText('-');
    const equalsButton = screen.getByText('=');

    fireEvent.click(screen.getByText('9'));
    fireEvent.click(minusButton);
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(equalsButton);
    expect(output.textContent).toBe('6');
  });

  test('Multiplies correctly based on input', () => {
    render(<CalculatorUI />);
    const output = screen.getByTestId('calculator-output');
    expect(output.textContent).toBe('');

    const multiplyButton = screen.getByText('x');
    const equalsButton = screen.getByText('=');

    fireEvent.click(screen.getByText('9'));
    fireEvent.click(multiplyButton);
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(equalsButton);
    expect(output.textContent).toBe('27');
  });

  test('Divides correctly based on input', () => {
    render(<CalculatorUI />);
    const output = screen.getByTestId('calculator-output');
    expect(output.textContent).toBe('');

    const addButton = screen.getByText('รท');
    const equalsButton = screen.getByText('=');

    fireEvent.click(screen.getByText('9'));
    fireEvent.click(addButton);
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(equalsButton);
    expect(output.textContent).toBe('3');
  });

  test('Performs Modulus Calculation correctly based on input', () => {
    render(<CalculatorUI />);
    const output = screen.getByTestId('calculator-output');
    expect(output.textContent).toBe('');

    const addButton = screen.getByText('%');
    const equalsButton = screen.getByText('=');

    fireEvent.click(screen.getByText('9'));
    fireEvent.click(addButton);
    fireEvent.click(screen.getByText('6'));
    fireEvent.click(equalsButton);
    expect(output.textContent).toBe('3');
  });
});
