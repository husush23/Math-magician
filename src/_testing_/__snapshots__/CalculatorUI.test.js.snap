// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="calculator"
>
  <div
    className="output"
    title="Output"
  >
    <span />
    <span />
    <span />
  </div>
  <div
    className="keyboard"
  >
    <button
      onClick={[Function]}
      type="submit"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      %
    </button>
    <button
      className="side"
      onClick={[Function]}
      type="submit"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      9
    </button>
    <button
      className="side"
      onClick={[Function]}
      type="submit"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      6
    </button>
    <button
      className="side"
      onClick={[Function]}
      type="submit"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      3
    </button>
    <button
      className="side"
      onClick={[Function]}
      type="submit"
    >
      +
    </button>
    <button
      className="span"
      onClick={[Function]}
      type="submit"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="submit"
    >
      .
    </button>
    <button
      className="side"
      onClick={[Function]}
      type="submit"
    >
      =
    </button>
  </div>
</div>
`;
